### PR TITLE
refactor(log): refactor log codebase to adhere to singe-export files

### PR DESCRIPTION
**Pull request by eryue0220**
This PR is to solve #4174.

### CODE REVIEW

File: log/mod.ts [hunk 1]
```diff
@@ -418,183 +387,10 @@ const state = {
...
-}
-
-setup(DEFAULT_CONFIG);
+export { critical } from "./critical.ts";
+export { debug } from "./debug.ts";
+export { error } from "./error.ts";
+export { getLogger } from "./get_logger.ts";
+export { info } from "./info.ts";
+export { setup } from "./_state.ts";
+export { warn } from "./warn.ts";
```
Comment (iuioiua) [inline log/mod.ts:396]: Can you please group exports together? There's another group above. Also, please use `export * from ...` in such a way that the exports from `mod.ts` are identical to how they were before.

File: log/mod.ts [hunk 2]
```diff
@@ -418,183 +387,10 @@ const state = {
...
-export function warning<T>(
-  msg: (T extends GenericFunction ? never : T) | (() => T),
-  ...args: unknown[]
-): T | undefined {
-  // Assist TS compiler with pass-through generic type
-  if (msg instanceof Function) {
-    return warn(msg, ...args);
-  }
-  return warn(msg, ...args);
-}
```
Comment (iuioiua) [inline log/mod.ts:504]: Where did these exports go? They should probably live in `log/warn.ts` until they're deprecated.

File: log/_state.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import { BaseHandler } from "./base_handler.ts";
+import { DEFAULT_CONFIG, DEFAULT_LEVEL } from "./_config.ts";
+import { type LogConfig, Logger } from "./logger.ts";
+
+export const state = {
+  handlers: new Map<string, BaseHandler>(),
+  loggers: new Map<string, Logger>(),
+  config: DEFAULT_CONFIG,
+};
```
Comment (iuioiua) [inline log/_state.ts:11]: Can you please move this to a separate `_state.ts` file and rename this current file `setup.ts`?

**Review by iuioiua — CHANGES_REQUESTED** [review]

File: log/mod.ts (hunk 1 above)
Comment (eryue0220) [reply log/mod.ts]: Updated.

File: log/_state.ts (hunk 3 above)
Comment (eryue0220) [reply log/_state.ts:11]: Updated.

File: log/mod.ts (hunk 2 above)
Comment (eryue0220) [reply log/mod.ts:504]: I thought it can be removed together. 

Updated.

**Pull request merged**